### PR TITLE
Feature/docker compose

### PR DESCRIPTION
**Pull request by VNaas**
the initial setup of docker is complete

### CODE REVIEW

**Review by chrstrom — APPROVED** [review]
I added some changes that are required for smooth operations, and cleaned up a few other bits. Feel free to merge once you also approve of the changes I made :smile: It might be a good idea for you to go over the changes as well @ronjakr

File: README.md [hunk 1]
```diff
@@ -7,11 +7,37 @@ Docker is a tool for creating a container, in this case running the ROS noetic i
...
-sudo docker-compose build
+$ sudo docker-compose build
 '''
+Note that you only need to build the image once unless changes have been made to the Dockerfile since the last time you built.
 After building the image, you can run any service using
 '''
-sudo docker-compose up -d <service_name>
+$ sudo docker-compose up -d <service_name>
 '''
+Running without a service name runs every container in the docker-compose file.
```
Comment (VNaas) [inline README.md:17]: Not sure if this is entirely correct. Docker documentation sucks

Comment (chrstrom) [reply README.md:17]: Yes and no. The compose file lists services which will be made into images when building. Not sure if there is a way of spinning up all images that is registered.

**Review by chrstrom — APPROVED** [review]
Things are looking good :smile: Some minor spelling mistakes in the README that I'll get to, but after that you can go ahead and merge.

It may also be a good idea to add a short section on how you can use aliases to avoid having to write those long-ass commands every time you want to spin up a container!

**Comment by chrstrom** [comment]
Also for the README, note that you can use the "Markdown: Open preview to the side" in vscode to make sure formatting is correct.

File: README.md [hunk 2]
```diff
@@ -3,41 +3,54 @@ Vortex ASV software. Purpose built for competing in ASV competitions.
...
+```
+
+or alternatively, it will source automatically the next time you run the container. To exit the container, run
+
+```
+exit
+```  
+
+## Volumes
+A volume is a folder which is linked between the container and the host the container the running on. The folder asw_ws in this repository is a volume, meaning any changes made inside the folder is automatically synced. /home/vortex/asv_ws/ is a volume, meaning making any changes in the volume from the container also changes the asv_ws folder in the host.
```
Comment (VNaas) [inline README.md:54]:
```suggestion
A volume is a folder which is linked between the container and the host the container the running on.  /home/vortex/asv_ws/ is a volume, meaning making any changes in the volume from the container also changes the asv_ws folder in the host. 
```

**Pull request merged**